### PR TITLE
card Angle by pam

### LINKED ISSUE #15: Create an Angle Value Object
Create a class in Domain called Angle.

Give it a single constructor that requires a double representing the angle to be provided. Make sure that this number is always a positive number between 0 and 360. If it is not, coerce it into a value between 0 and 360. Create unit tests to confirm this functionality for a variety of numbers.

Let its value field be private final. (Final makes it not writable outside of the constructor)

Provide a getter for that value.

See UML for details

image

### LINKED ISSUE #24: Create an Owner Class
resolves #24 owner class - pam
Create a class named Owner in Domain and implement it according to this UML. Let private fields be final.

image

### LINKED ISSUE #26: Create a MoveTankCommand class
Create a MoveTankCommand class according to this UML. The private fields should be final.

image

### PR DESCRIPTION
**Pull request by pvfpinho**
resolves #15 - by pam...Aaron I think My private final is at the wrong place... Thanks  
resolves #24 owner class - pam
resolves #26 class MoveTankCommand - pam

### CODE REVIEW

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Let its value field be private final. (Final makes it not writable outside of the constructor)
+ *
+ * Provide a getter for that value.
+ *
+ * See UML for details
+ */
+public class Angle;
+//class
+{
+    private final doble angle;//not sure if that is at the right place, intended to make it not writable
```
Comment (ajshiff) [inline domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java]: double*

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/**
```
Comment (ajshiff) [inline domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java:2]: need package name. `com.jakefoundation.buildabotworkshop.domain`

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+/* Create a class named Owner in Domain and implement it according to this UML.
+ * Let private fields be final.
+ * */
+
+public class Owner {
```
Comment (ajshiff) [inline domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java:3]: looks like this also resolves #24. Please add that to your PR description :)

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (ajshiff) [inline domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java]: this file also needs package name. `com.jakefoundation.buildabotworkshop.domain`

**Review by ajshiff — CHANGES_REQUESTED** [review]

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+//class
+{
+    private final doble angle;//not sure if that is at the right place, intended to make it not writable
+
+public Angle(double angleValue) {
+        //constructor
+        this.angle = angleValue % 360;
+        if (this.angle > 0)
+        {
+        this.angle -= 360;
```
Comment (ajshiff) [inline domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java]: this looks like it is coercing the value to always be negative. Did you mean `this.angle < 0` and `this.angle += 360`

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java (hunk 1 above)
Comment (pvfpinho) [reply domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java]: okay

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java (hunk 2 above)
Comment (pvfpinho) [reply domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Angle.java:2]: okay

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java (hunk 3 above)
Comment (pvfpinho) [reply domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java:3]: how do I do that again?

File: domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java (hunk 4 above)
Comment (pvfpinho) [reply domain/src/main/java/com/jakefoundation/buildabotworkshop/domain/Owner.java]: okay

**Review by ajshiff — APPROVED** [review]
All needed changes have been made. :)

**Pull request merged**